lepton/azure_network.go

```diff
@@ -274,30 +280,37 @@ func (a *Azure) getNsgClient() network.SecurityGroupsClient {
 
 // CreateNetworkSecurityGroup creates a new network security group with
 // rules set for allowing SSH and HTTPS use
-func (a *Azure) CreateNetworkSecurityGroup(ctx context.Context, location string, nsgName string) (nsg network.SecurityGroup, err error) {
+func (a *Azure) CreateNetworkSecurityGroup(ctx context.Context, location string, nsgName string, c *Config) (nsg network.SecurityGroup, err error) {
 	nsgClient := a.getNsgClient()
+
+	securityRules := make([]network.SecurityRule, len(c.RunConfig.Ports))
+
+	for i, port := range c.RunConfig.Ports {
+
+		var rule = network.SecurityRule{
+			Name: to.StringPtr("allow_" + strconv.Itoa(port)),
+			SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
+				Protocol:                 network.SecurityRuleProtocolTCP,
+				SourceAddressPrefix:      to.StringPtr("0.0.0.0/0"),
+				SourcePortRange:          to.StringPtr("1-65535"),
+				DestinationAddressPrefix: to.StringPtr("0.0.0.0/0"),
+				DestinationPortRange:     to.StringPtr(strconv.Itoa(port)),
+				Access:                   network.SecurityRuleAccessAllow,
+				Direction:                network.SecurityRuleDirectionInbound,
+				Priority:                 to.Int32Ptr(rand.Int31n(1000-100) + 100), //Generating number between 100 - 1000
```

Comment:
should this just be set to 1000 (which i think is 'low priority')

Reply:
Multiple rules can't have same priority. I tried.